components/org.wso2.carbon.micro.integrator.extensions/org.wso2.carbon.micro.integrator.management.apis/src/main/java/org/wso2/carbon/micro/integrator/management/apis/MessageProcessorResource.java

```diff
@@ -0,0 +1,252 @@
+        if (messageProcessor instanceof ScheduledMessageForwardingProcessor) {
+            return SCHEDULED_MESSAGE_FORWARDING_TYPE;
+        } else if (messageProcessor instanceof FailoverScheduledMessageForwardingProcessor) {
+            return FAILOVER_SCHEDULED_MESSAGE_FORWARDING_TYPE;
+        } else if (messageProcessor instanceof SamplingProcessor) {
+            return SAMPLING_PROCESSOR_TYPE;
+        } else {
+            return CUSTOM_PROCESSOR_TYPE;
+        }
+    }
+
+    /**
+     * Changes the processor state to the specified state
+     * @param messageContext Axis2message context
+     * @param jsonObject message payload
+     * */
+    private void changeProcessorStatus(MessageContext messageContext, JsonObject jsonObject) {
+        org.apache.axis2.context.MessageContext axis2MessageContext =
+                ((Axis2MessageContext) messageContext).getAxis2MessageContext();
+        SynapseConfiguration synapseConfiguration = messageContext.getConfiguration();
+        String processorName = jsonObject.get(PROCESSOR_NAME_PARAM).getAsString();
+        String status = jsonObject.get(STATUS_PROPERTY).getAsString();
+
+        MessageProcessor messageProcessor = synapseConfiguration.getMessageProcessors().get(processorName);
+        if (Objects.nonNull(messageProcessor)) {
+            JSONObject jsonResponse = new JSONObject();
+            if (PROCESSOR_INACTIVE_STATUS.equalsIgnoreCase(status)) {
+                messageProcessor.deactivate();
+                jsonResponse.put("Message", processorName + " : is deactivated");
+                Utils.setJsonPayLoad(axis2MessageContext, jsonResponse);
+                return;
+            }
+            if (PROCESSOR_ACTIVE_STATUS.equalsIgnoreCase(status)) {
+                messageProcessor.activate();
+                jsonResponse.put("Message", processorName + " : is activated");
+                Utils.setJsonPayLoad(axis2MessageContext, jsonResponse);
+                return;
+            }
+        } else {
+            LOG.warn("Specified message processor does not exist");
+            Utils.setJsonPayLoad(axis2MessageContext, Utils.createJsonErrorObject("Message processor does not exist"));
+        }
+
+    }
+
+    /** Returns the JSON payload of a given message
+     * @param messageContext synapseMessageContext
+     * @return JsonObject payload
+     * */
+    private JsonObject getJsonPayload(MessageContext messageContext) throws IOException {
+        org.apache.axis2.context.MessageContext axis2MessageContext =
+                ((Axis2MessageContext) messageContext).getAxis2MessageContext();
+        InputStream jsonStream = JsonUtil.getJsonPayload(axis2MessageContext);
+        String jsonString = IOUtils.toString(jsonStream);
+        return new JsonParser().parse(jsonString).getAsJsonObject();
+    }
+
+    /**
+     * Returns the state of the MessageProcessor
+     * @param isDeactivated
```

Comment:
Shall we add a description? This is shown as a warning in the IDE.

Reply:
done